instrumentation/src/main/java/io/opentelemetry/android/instrumentation/startup/InitializationListener.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation.startup
+
+import io.opentelemetry.api.OpenTelemetry
+
+/**
+ * Provides callbacks to know the sate of the initialization.
+ */
+interface InitializationListener {
+    /**
+     * Called when the RUM initialization starts.
+     */
+    fun onStart()
+
+    /**
+     * Called when the RUM initialization ends.
+     * @param openTelemetry - The initialized OpenTelemetry instance.
+     */
+    fun onEnd(openTelemetry: OpenTelemetry)
```

Comment:
What if this was `onEnd(rum: OpenTelemetryRum)` instead? The otel instance (like what you have here) as well as the session id are both obtainable from the `OpenTelemetryRum` instance.

Reply:
Makes sense, though now that I think about the whole thing after [the proposal](https://github.com/open-telemetry/opentelemetry-android/issues/214) I wrote yesterday, it seems to me that these hooks might overlap with [instrumentation installers](https://github.com/open-telemetry/opentelemetry-android/blob/d7d5045c8178b0a6069ae93c1cd0b11dc6641fe0/instrumentation/src/main/java/io/opentelemetry/android/instrumentation/InstrumentedApplication.java#L19). So maybe it's not needed to add another, similar mechanism if we decide to make that one public?